components/bsx/Create/CreateToken.vue

```diff
@@ -66,258 +69,255 @@
   }
-  protected metadata = ''
-  protected balanceNotEnough = false
-  @Ref('balanceInput') readonly balanceInput
-  @Ref('baseTokenForm') readonly baseTokenForm
+)
 
-  @Watch('price')
-  protected updatePrice(value: string) {
-    this.price = value
-    this.balanceInput.checkValidity()
-  }
+const { apiUrl } = useApi()
+const { urlPrefix, tokenId } = usePrefix()
+const { accountId } = useAuth()
+const {
+  status: transactionStatus,
+  isLoading: isTransactionLoading,
+  stopLoader: stopTransactionLoader,
+} = useTransactionStatus()
 
-  get balanceOfToken() {
-    return getBalance(this.feesToken)
-  }
-  get depositOfToken() {
-    return getDeposit(this.feesToken, parseFloat(this.deposit))
-  }
+onApiConnect(apiUrl.value, (api) => {
+  const instanceDeposit = getInstanceDeposit(api)
+  const metadataDeposit = getMetadataDeposit(api)
+  deposit.value = (instanceDeposit + metadataDeposit).toString()
+})
 
-  get balanceNotEnoughMessage() {
-    if (this.balanceNotEnough) {
-      return this.$t('tooltip.notEnoughBalance')
-    }
-    return ''
-  }
+const base = ref<BaseTokenType>({
+  name: '',
+  file: null,
+  description: '',
+  selectedCollection: null,
+  edition: 1,
+  secondFile: null,
+})
+const collections = ref<MintedCollection[]>([])
+const postfix = ref(true)
+const deposit = ref('0')
+const attributes = ref<Attribute[]>([])
+const nsfw = ref(false)
+const price = ref('0')
+const listed = ref(false)
+const hasRoyalty = ref(true)
+const feesToken = ref<Token>('BSX')
+const royalty = ref<Royalty>({
+  amount: 0.15,
+  address: '',
+})
+const balanceNotEnough = ref(false)
 
-  public async created() {
-    onApiConnect(this.apiUrl, (api) => {
-      const instanceDeposit = getInstanceDeposit(api)
-      const metadataDeposit = getMetadataDeposit(api)
-      this.deposit = (instanceDeposit + metadataDeposit).toString()
-    })
+const balanceInput = ref<typeof TokenBalanceInputComponent | null>(null)
+const baseTokenForm = ref<typeof BaseTokenFormComponent | null>(null)
+
+watch(price, (value) => {
+  price.value = value
+  balanceInput.value?.checkValidity()
+})
+
+const balanceOfToken = computed(() => getBalance(feesToken.value))
+const depositOfToken = computed(() =>
+  getDeposit(feesToken.value, parseFloat(deposit.value))
+)
+const balanceNotEnoughMessage = computed(() => {
+  if (balanceNotEnough.value) {
+    return $i18n.t('tooltip.notEnoughBalance')
   }
+  return ''
+})
 
-  @Watch('accountId', { immediate: true })
-  async hasAccount(value: string, oldVal: string) {
-    if (shouldUpdate(value, oldVal)) {
-      this.fetchCollections()
-      this.feesToken = await getFeesToken()
+const loadCollectionMeta = async () => {
+  const metadata = collections.value.map(({ metadata }) => metadata)
+
+  metadata.forEach(async (m, i) => {
+    try {
+      const meta = await fetchCollectionMetadata(collections[i])
+      set(collections.value, i, {
+        ...collections.value[i],
+        ...meta,
+      })
```

Comment:
use spread operator or make it reactive

Reply:
I gave reactivity to the meta properties by setting their initial value to `collections.value`